manifests/install.pp

```diff
@@ -1,15 +1,15 @@
 # Install the foreman proxy
 class foreman_proxy::install {
   if ! $foreman_proxy::custom_repo {
-    foreman::repos { 'foreman_proxy':
+    class { '::foreman::repo':
```

Comment:
That depends on the parse order: if puppet-foreman_proxy is parsed before puppet-foreman, it will work.